src/docs/ad/ad-service-discovery.rst

```diff
@@ -0,0 +1,87 @@
+Service Discovery
+#################
+
+Active Directory Terms
+======================
+
+`Active Directory <https://en.wikipedia.org/wiki/Active_Directory>`_ is a directory service providing Windows and Linux clients with information about centrally stored user and group information. Below are some Active Directory terms that are important to understand for the purposes of this document.
+
+Domain Controller
+    The Active Directory Domain Controller (DC for short) is an important source of user information and authentication in many computing environments. The DC stores user and group information in a LDAP directory server (database) and provides this information as a `service <https://docs.microsoft.com/en-us/windows-server/identity/ad-ds/get-started/virtual-dc/active-directory-domain-services-overview>`_. This allows administrators to configure clients to retrieve user information from the AD DC. Similarly, the DC contains Kerberos data used to authenticate users password and access to other network services.
+
+Domain
+    A domain is defined as a logical group of network objects (computers, users, devices) that share the same Active Directory database.
+
+Forest
+    A forest is a collection of trees that share a common global catalog, directory schema, logical structure, and directory configuration. A forest contains one or multiple domains.
+
+Site
+    A logical organization, or container for objects within a domain. `Sites <https://docs.microsoft.com/en-us/windows-server/identity/ad-ds/plan/understanding-active-directory-site-topology>`_ are often used to group geographically-close systems to together. Clients can prefer to communicate only to servers within a site.
+
+Global Catalog
+    A service advertising information about Active Directory domains in a forest, it can be searched by clients similarly to normal LDAP user/group information. The `Global catalog <https://docs.microsoft.com/en-us/windows/win32/ad/global-catalog>`_ is used by SSSD clients to read and discovery AD Site and Forest information, and trusted domains.
+
+CLDAP Ping
+    See section below
+
+When using the AD provider, by default SSSD clients utilize auto-discovery of Active Directory servers to help avoid explicitly configuring Active Directory servers, and enable failover behavior. The term **Service Discovery** may also be used in SSSD documentation and manual pages to describe this.
```

Comment:
"Active Directory servers to help avoid explicitly configuring Active Directory servers" -> "Active Directory servers to avoid the need to explicitly define them on sssd.conf on each client."?

Also explain the goal? Ie to discover global catalog and domain controlles that will be used as primary and backup servers. Little explanation of primary servers (in-site) and backup (off-site when primary are not available)?